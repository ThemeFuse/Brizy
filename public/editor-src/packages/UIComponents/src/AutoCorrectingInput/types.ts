import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
