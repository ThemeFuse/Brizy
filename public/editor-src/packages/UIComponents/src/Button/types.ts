import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
