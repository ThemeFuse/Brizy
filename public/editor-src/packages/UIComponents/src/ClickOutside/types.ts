import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
