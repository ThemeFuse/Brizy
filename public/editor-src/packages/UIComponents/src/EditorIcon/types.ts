export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
