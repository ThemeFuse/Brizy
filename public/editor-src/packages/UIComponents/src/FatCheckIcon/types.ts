import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
