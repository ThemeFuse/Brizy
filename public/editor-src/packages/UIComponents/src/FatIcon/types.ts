import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
