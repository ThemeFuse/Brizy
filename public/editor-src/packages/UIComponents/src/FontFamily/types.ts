import { FontFamilyType } from "../types/fonts/familyType";
import { Weight } from "../types/fonts/weight";

export type FontsBlock = Partial<{
  config: Array<FontObject>;
  blocks: Array<FontObject>;
  [FontFamilyType.google]: Array<FontObject>;
  [FontFamilyType.upload]: Array<FontObject>;
}>;

export interface Props {
  addFont?: VoidFunction;
  onChange: (font: FontWithType) => void;
  value: string;
  className?: string;
  addFontLabel?: string;
  fonts: FontsBlock;
}
export interface FontSizes {
  default: string;
  great_vibes: string;
  alex_brush: string;
  allura: string;
  parisienne: string;
}

export interface FontObject {
  brizyId?: string;
  family: string;
  id: string;
  title: string;
  weights: Array<Weight>;
}

export interface FontWithType {
  id: string;
  type: FontFamilyType;
  weights: Array<Weight>;
}

export interface FontList {
  fonts: Array<FontObject>;
  type: FontFamilyType;
}
