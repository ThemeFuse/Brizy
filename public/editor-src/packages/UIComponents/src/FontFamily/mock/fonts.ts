import { FontsBlock } from "../types";

export const fonts: FontsBlock = {
  blocks: [
    {
      id: "noto_serif",
      title: "Noto Serif",
      family: "Noto Serif, serif",
      weights: [400, 700],
      brizyId: "uemcgmfyliwunxbjbcmeqhnxnidrfpbpebqf"
    },
    {
      id: "montserrat",
      title: "Montserrat",
      family: "Montserrat, sans-serif",
      weights: [400, 100, 200, 300, 500, 600, 700, 800, 900],
      brizyId: "iyunwhlktvgybtevnoutoiyvnujjvobsrgvp"
    }
  ],
  config: [
    {
      id: "lato",
      title: "Lato",
      family: "Lato, sans-serif",
      weights: [400, 100, 300, 700, 900],
      brizyId: "uzrpsocdxtgrkbxjjxkchqcybpvpzsuvdlji"
    },
    {
      id: "overpass",
      title: "Overpass",
      family: "Overpass, sans-serif",
      weights: [400, 100, 200, 300, 600, 700, 800, 900],
      brizyId: "qwhwsomltrpyogspgbomkxquvqsqfdlvcnfo"
    },
    {
      id: "red_hat_text",
      title: "Red Hat Text",
      family: "Red Hat Text, sans-serif",
      weights: [400, 500, 700],
      brizyId: "eytgthrgfzlrrzxlhynabspndabldgdbdjnm"
    },
    {
      id: "dm_serif_text",
      title: "DM Serif Text",
      family: "DM Serif Text, serif",
      weights: [400],
      brizyId: "pujmflqmocbjojknwlnidilgqedjzqftpnrv"
    },

    {
      id: "blinker",
      title: "Blinker",
      family: "Blinker, sans-serif",
      weights: [400, 100, 200, 300, 600, 700, 800, 900],
      brizyId: "yhkoopjikembswaygkzktfmiiashwjcrvbxr"
    },
    {
      id: "aleo",
      title: "Aleo",
      family: "Aleo, serif",
      weights: [400, 300, 700],
      brizyId: "ucgecsrbcjkpsfctgzwsocokuydcdgiubroh"
    },
    {
      id: "nunito",
      title: "Nunito",
      family: "Nunito, sans-serif",
      weights: [400, 200, 300, 600, 700, 800, 900],
      brizyId: "ppzycxqtiwtmjnfpbfluoynrnnfviuerjczz"
    }
  ]
};
