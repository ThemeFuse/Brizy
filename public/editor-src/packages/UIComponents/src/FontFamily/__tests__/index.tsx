import { fireEvent, render, waitFor } from "@testing-library/react";
import React, { useState } from "react";
import { FontFamily } from "..";
import { fonts } from "../mock/fonts";
import { FontWithType } from "../types";

const MockComponent = () => {
  const [value, setValue] = useState("overpass");
  const addFont = () => undefined;

  const onChange = (value: FontWithType) => {
    setValue(value.id);
  };

  return (
    <FontFamily
      addFontLabel="Add New Font"
      addFont={addFont}
      onChange={onChange}
      value={value}
      fonts={fonts}
    />
  );
};

describe("Test FontFamily", () => {
  const addFont = jest.fn();
  const onChange = jest.fn();

  describe("Snapshots", () => {
    test("Render font list", async () => {
      const { container } = render(
        <FontFamily
          addFontLabel="Add New Font"
          addFont={addFont}
          onChange={onChange}
          value="overpass"
          fonts={fonts}
        />
      );
      const current = container.firstElementChild;

      await waitFor(() => {
        expect(current).toBeInTheDocument();
      });

      expect(current).toMatchSnapshot();
    });
    test("Render without addFontLabel prop", async () => {
      const { container } = render(
        <FontFamily
          addFont={addFont}
          onChange={onChange}
          value="overpass"
          fonts={fonts}
        />
      );
      const current = container.firstElementChild;

      await waitFor(() => {
        expect(current).toBeInTheDocument();
      });

      const typographyAdder = container.querySelector(
        ".brz-ed-font__typography-adder"
      );

      expect(typographyAdder).toBeInTheDocument();
      expect(current).toMatchSnapshot();
    });
  });

  test("Checking if active value has class", async () => {
    const { container } = render(<MockComponent />);

    const current = container.querySelector(".brz-ed-font__name");

    await waitFor(() => {
      expect(current).toBeInTheDocument();
    });

    if (current) {
      fireEvent.click(current, {
        value: "noto_serif"
      });
    }

    expect(current).not.toBeNull();
    expect(current).toHaveClass("active");
  });

  test("Checking addFont function", async () => {
    const { container } = render(
      <FontFamily
        addFont={addFont}
        onChange={onChange}
        value="overpass"
        fonts={fonts}
      />
    );
    const current = container.querySelector(".brz-ed-font__typography-adder");

    await waitFor(() => {
      expect(current).toBeInTheDocument();
    });

    current && fireEvent.click(current);

    expect(addFont).toHaveBeenCalledTimes(1);
  });
});
