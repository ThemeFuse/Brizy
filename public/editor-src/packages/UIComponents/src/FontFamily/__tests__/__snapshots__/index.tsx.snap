// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FontFamily Snapshots Render font list 1`] = `
<div
  class="brz-ed-font__typography"
>
  <div
    style="position: relative; overflow: hidden; width: 100%; height: 100%;"
  >
    <div
      style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; overflow: scroll; margin-right: 0px; margin-bottom: 0px;"
    >
      <div
        class="brz-ed-font__name"
        style="font-family: Noto Serif, serif; font-size: 17px;"
      >
        Noto Serif
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Montserrat, sans-serif; font-size: 17px;"
      >
        Montserrat
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Lato, sans-serif; font-size: 17px;"
      >
        Lato
      </div>
      <div
        class="brz-ed-font__name active"
        style="font-family: Overpass, sans-serif; font-size: 17px;"
      >
        Overpass
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Red Hat Text, sans-serif; font-size: 17px;"
      >
        Red Hat Text
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: DM Serif Text, serif; font-size: 17px;"
      >
        DM Serif Text
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Blinker, sans-serif; font-size: 17px;"
      >
        Blinker
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Aleo, serif; font-size: 17px;"
      >
        Aleo
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Nunito, sans-serif; font-size: 17px;"
      >
        Nunito
      </div>
    </div>
    <div
      style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
    >
      <div
        class="brz-scrollColor__dark"
        style="position: relative; display: block; height: 100%; border-radius: inherit;"
      />
    </div>
    <div
      style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
    >
      <div
        class="brz-scrollColor__dark"
        style="position: relative; display: block; width: 100%; border-radius: inherit;"
      />
    </div>
  </div>
  <div
    class="brz-ed-font__typography-adder"
  >
    <svg
      class="brz-icon-svg brz-ed-icon-svg align-[initial]"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <g>
        <path
          d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
          fill="currentColor"
        />
      </g>
    </svg>
    Add New Font
  </div>
</div>
`;

exports[`Test FontFamily Snapshots Render without addFontLabel prop 1`] = `
<div
  class="brz-ed-font__typography"
>
  <div
    style="position: relative; overflow: hidden; width: 100%; height: 100%;"
  >
    <div
      style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; overflow: scroll; margin-right: 0px; margin-bottom: 0px;"
    >
      <div
        class="brz-ed-font__name"
        style="font-family: Noto Serif, serif; font-size: 17px;"
      >
        Noto Serif
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Montserrat, sans-serif; font-size: 17px;"
      >
        Montserrat
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Lato, sans-serif; font-size: 17px;"
      >
        Lato
      </div>
      <div
        class="brz-ed-font__name active"
        style="font-family: Overpass, sans-serif; font-size: 17px;"
      >
        Overpass
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Red Hat Text, sans-serif; font-size: 17px;"
      >
        Red Hat Text
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: DM Serif Text, serif; font-size: 17px;"
      >
        DM Serif Text
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Blinker, sans-serif; font-size: 17px;"
      >
        Blinker
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Aleo, serif; font-size: 17px;"
      >
        Aleo
      </div>
      <div
        class="brz-ed-font__name"
        style="font-family: Nunito, sans-serif; font-size: 17px;"
      >
        Nunito
      </div>
    </div>
    <div
      style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
    >
      <div
        class="brz-scrollColor__dark"
        style="position: relative; display: block; height: 100%; border-radius: inherit;"
      />
    </div>
    <div
      style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
    >
      <div
        class="brz-scrollColor__dark"
        style="position: relative; display: block; width: 100%; border-radius: inherit;"
      />
    </div>
  </div>
  <div
    class="brz-ed-font__typography-adder"
  >
    <svg
      class="brz-icon-svg brz-ed-icon-svg align-[initial]"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <g>
        <path
          d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
          fill="currentColor"
        />
      </g>
    </svg>
    Add New Font
  </div>
</div>
`;
